frontend/catalyst/passes/builtin_passes.py

```diff
@@ -717,85 +700,75 @@ def circuit():
+        This pass is only for analysis with the ``null.qubit`` device and potential future execution
+        when a suitable backend is available.
 
     Args:
         fn (QNode): QNode to apply the pass to
-        max_pauli_size (int): The maximum size of the Pauli strings after merging.
+        max_pauli_size (int):
+            The maximum size of Pauli strings resulting from merging. If a merge results in a PPM
+            that acts on more than ``max_pauli_size`` qubits, that merge will not be performed. The
+            default value is ``0`` (no limit).
 
     Returns:
         :class:`QNode <pennylane.QNode>`
 
     **Example**
 
-    In this example, the Clifford+T gates will be converted into PPRs first,
-    then the Clifford PPRs will be commuted past the non-Clifford PPR,
-    and finally the Clifford PPRs will be absorbed into the Pauli Product Measurements.
+    The ``merge_ppr_ppm`` compilation pass can be applied as a dectorator on a QNode:
 
     .. code-block:: python
 
         import pennylane as qml
-        from catalyst import qjit, measure
+        from functools import partial
+        import jax.numpy as jnp
 
-        ppm_passes = [("PPM",["to-ppr", "commute-ppr","merge-ppr-ppm",])]
+        qml.capture.enable()
 
-        @qjit(pipelines=ppm_passes, keep_intermediate=True, target="mlir")
-        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        @qml.qjit(target="mlir")
+        @partial(qml.transforms.merge_ppr_ppm, max_pauli_size=2)
+        @qml.qnode(qml.device("null.qubit", wires=2))
         def circuit():
-            qml.H(0)
-            qml.T(0)
-            return measure(0)
-
-        print(circuit.mlir_opt)
-
-    Example MLIR Representation:
-
-    .. code-block:: mlir
-
-        . . .
-        %2 = qec.ppr ["X"](8) %1 : !quantum.bit
-        %mres, %out_qubits = qec.ppm ["X"] %2 : !quantum.bit
-        . . .
-
-    If a merging resulted in a PPM acting on more than
-    `max_pauli_size` qubits (here, `max_pauli_size = 2`), that merging would be skipped.
-
-    .. code-block:: python
-
-        from catalyst import measure, qjit
-        from catalyst.passes import to_ppr, merge_ppr_ppm
+            qml.PauliRot(jnp.pi / 2, pauli_word="Z", wires=0)
+            qml.PauliRot(jnp.pi / 2, pauli_word="X", wires=0)
+            qml.PauliRot(jnp.pi / 2, pauli_word="Z", wires=0)
 
-        pips = [("pipe", ["enforce-runtime-invariants-pipeline"])]
+            ppm = qml.pauli_measure(pauli_word="ZX", wires=[0, 1])
 
-        @qjit(pipelines=pips, target="mlir")
-        @to_ppr
-        @merge_ppr_ppm(max_pauli_size=2)
-        @qml.qnode(qml.device("lightning.qubit", wires=3))
-        def circuit():
-            qml.CNOT([1, 2])
-            qml.CNOT([0, 1])
-            qml.CNOT([0, 2])
-            return measure(0), measure(1), measure(2)
+            qml.PauliRot(jnp.pi / 4, pauli_word="Z", wires=0)
```

Comment:
@isaacdevlugt This won't achieve the expected `merging` behaviour. To merge, you need the final operation to be a `ppm`.  I suggest removing the last `qml.PauliRot(...)` and rerunning this. It should give a different result.